dms-service/src/main/java/de/muenchen/oss/swim/dms/application/usecase/ProcessFileUseCase.java

```diff
@@ -193,14 +194,42 @@ protected DmsTarget resolveMetadataTargetCoo(final UseCase.Type resourceType, fi
         }
         // extract coo and username from metadata
         final DmsTarget metadataTarget = switch (resourceType) {
-        case INBOX -> dmsMetadataHelper.resolveInboxDmsTarget(metadata);
-        case INCOMING_OBJECT -> dmsMetadataHelper.resolveIncomingDmsTarget(metadata);
+        case INBOX_CONTENT_OBJECT -> dmsMetadataHelper.resolveInboxDmsTarget(metadata);
+        case PROCEDURE_INCOMING -> dmsMetadataHelper.resolveIncomingDmsTarget(metadata);
         case METADATA_FILE -> throw new IllegalStateException("Target type metadata needs to be resolved to other types");
         };
         // combine resolves target with use case
         return this.combineDmsTargetWithUseCase(metadataTarget, useCase);
     }
 
+    /**
+     * Resolve target coo via dms object name.
+     *
+     * @param resourceType The resource type of the target.
+     * @param metadata The metadata used for resolving the name pattern.
+     * @param useCase The use case to resolve the target for.
+     * @param file The file to resolve the target for.
+     * @return Dms target resolved via dms object name.
+     */
+    protected DmsTarget resolveTargetCooViaName(final DmsResourceType resourceType, final Metadata metadata, final UseCase useCase, final File file) {
+        // validate required use case properties
+        if (Strings.isBlank(useCase.getFilenameNamePattern())) {
+            throw new IllegalArgumentException("DMS target coo via object name: Filename name pattern is required");
+        }
+        if (Strings.isBlank(useCase.getUsername())) {
+            throw new IllegalStateException("DMS target coo via object name: Username is required");
+        }
+        // resolve lookup name
+        final String objectName = this.patternHelper.applyPattern(useCase.getFilenameNamePattern(), file.getFileName(), metadata);
+        // search for object name
+        final DmsTarget requestContext = new DmsTarget(null, useCase.getUsername(), useCase.getJoboe(), useCase.getJobposition());
+        final List<String> coos = this.dmsOutPort.findObjectsByName(resourceType, objectName, requestContext);
+        if (coos.size() != 1) {
+            throw new IllegalStateException(String.format("DMS target coo via object name: Found %d instead of exactly one object", coos.size()));
```

Comment:
Can we maybe also print the pattern we are looking for (i.e. `objectName`) into the error message? Would make it easier to find the problem...